### PR TITLE
feat: add ica v3 docs

### PR DESCRIPTION
**Pull request by aroralanuk**

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -188,6 +188,11 @@ const sidebars = {
           id: "reference/applications/warp-routes",
           label: "Warp Routes",
         },
+        {
+          type: "doc",
+          id: "reference/applications/ica",
+          label: "InterchainAccount",
```
Comment (paulbalaji) [inline sidebars.js:194]:
```suggestion
          label: "Interchain Accounts",
```

File: docs/reference/applications/ica.mdx [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Interchain Account Interface
+
+Developers can use the **Interchain Account** interface to create and control an account on a remote chain from their local chain.
+
+Unlike general message passing, which requires recipients to implement a specific interface, Interchain Accounts (ICAs) allow developers to interact with _any remote_ contract.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:5]:
```suggestion
Unlike general message passing, which requires recipients to implement a specific interface, Interchain Accounts (ICAs) allow developers to interact with _any_ remote contract.
```

File: docs/reference/applications/ica.mdx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    click A_D https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/middleware/InterchainAccountRouter.sol
+
+    style Sender fill:#FF0099
+    style SenderAccount fill:#FF0099
+    style Recipient fill:#FF0099
+
+
+```
+
+Interchain Accounts allow you to make a remote call from chain A to chain B using the router (`**InterchainAccountRouter**`). Here's how it works:
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:47]:
```suggestion
Interchain Accounts allow you to make a remote call from **Chain A** to **Chain B** using the router (`InterchainAccountRouter`). Here's how it works:
```

File: docs/reference/applications/ica.mdx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+- We use [CREATE2](https://docs.openzeppelin.com/cli/2.8/deploying-with-create2) to compute the determinstic [OwnableMulticall](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/middleware/libs/OwnableMulticall.sol) contract address for you which serves as a proxy for your cross-chain calls. You can do it yourself too [here](#computing-addresses).
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:49]:
```suggestion
- We use [CREATE2](https://docs.openzeppelin.com/cli/2.8/deploying-with-create2) to compute the determinstic [OwnableMulticall](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/middleware/libs/OwnableMulticall.sol) contract address for you, which serves as a proxy for your cross-chain calls. You can do it yourself too [here](#computing-addresses).
```

File: docs/reference/applications/ica.mdx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+- You can encode your call which includes the to address, call data, and the msg.value for each call, batched together in a array.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:51]:
```suggestion
- You can encode your call which includes the to address, call data, and the `msg.value` for each call, batched together in a array.
```

File: docs/reference/applications/ica.mdx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+- You send the encoded call to the chainA router which gets relayed to chainB router.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:53]:
```suggestion
- You send the encoded call to the **Chain A** router which gets relayed to the **Chain B** router.
```

File: docs/reference/applications/ica.mdx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+- After decoding the calls, the ChainB router checks if the computed address is already deployed or not. If not, we deploy the _OwnableMulticall_ contract.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:55]:
```suggestion
- After decoding the calls, the **Chain B** router checks if the computed address is already deployed or not. If not, we deploy the _OwnableMulticall_ contract.
```

File: docs/reference/applications/ica.mdx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+- The router then performs a multicall on the ICA address which in turn makes the desired arbitary call on chainB.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:57]:
```suggestion
- The router then performs a multicall on the ICA address, which in turn makes the desired arbitary call on **Chain B**.
```

File: docs/reference/applications/ica.mdx [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+The Interchain Account interface assigns every `(uint32 origin, address owner, address remoteRouter, address remoteISM)` tuple a unique ICA address. The sender owns that ICA on the destination chain, and can direct it to make arbitrary function calls via the `InterchainAccountRouter.callRemote()` endpoint.
+
+On many chains, you are able to use the defaults that are set by the owner of the router contract. See the [#overrides](#overrides) section to see how to make calls to any chain.
```
Comment (paulbalaji) [inline docs/reference/applications/ica.mdx:61]:
```suggestion
For core chains supported by Hyperlane, you are able to use the defaults that are set by the owner of the router contract. See the [#overrides](#overrides) section to see how to make calls to _any_ chain.
```

**Review by yorhodes — COMMENTED** [review]
submitting a stale review

File: docs/reference/applications/ica.mdx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+```
+
+:::tip
+
+Want to use `InterchainAccountRouter` out of the box? Please refer to [addresses](../contract-addresses.mdx#interchainaccountrouter)
+
+:::
+
+## Example Usage
```
[the remark this answers is not in the document]
Comment (aroralanuk) [reply docs/reference/applications/ica.mdx:91]: won't match the rest of the docs, but I'll just link it as a tip.

File: docs/reference/applications/ica.mdx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+      SenderAccount
+      Recipient
+    end
+
+    Sender -- "callRemote(destination, recipient, call)" --> A_O
+    A_O -- "dispatch(destination, router, \n[sender, recipient, call])" --> M_O
+    M_O -. "relay" .- M_D
+    M_D -- "handle(origin, router, \n[sender, recipient, call])" --> A_D
+    A_D == "interchainAccount(origin, sender)" ==> SenderAccount
+    SenderAccount -- "call(data)" --> Recipient
```
[the remark this answers is not in the document]
Comment (aroralanuk) [reply docs/reference/applications/ica.mdx:35]: fixed in the next pr

File: docs/reference/applications/ica.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (aroralanuk) [reply docs/reference/applications/ica.mdx:91]: actually added a section

**Pull request merged**